catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/rtl/ShowTextBrickSupportNonLatinStringAndNumberTest.java

```diff
@@ -0,0 +1,244 @@
+	}
+
+	@Test
+	public void testShowPashtoNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ps"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowFarsiNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("fa"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowUrduNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ur"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowChineseSimplifiedNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("zh", "CN"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowChineseTraditionalNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("zh", "TW"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowKoreanNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ko"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowJapanNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ja"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowHindiNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("hi"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowGujaratiNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("gu"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowTamilNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ta"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowTeluguNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("te"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowThaiNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("th"), "%f", zeroToNine));
+	}
+
+	@Test
+	public void testShowMalayalamNumber() throws Exception {
+		printNonLatinStringOnStage(String.format(new Locale("ml"), "%f", zeroToNine));
+	}
+
+	private Script createProjectAndGetStartScriptWithImages(String projectName) {
+		Project project = new Project(null, projectName);
+		Sprite sprite = new Sprite("testSprite");
+		Script script = new StartScript();
+
+		sprite.addScript(script);
+
+		script.addBrick(new SetVariableBrick());
+		script.addBrick(new ShowTextBrick());
+
+		project.getDefaultScene().addSprite(sprite);
+
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+		return script;
+	}
+
+	private void printNonLatinStringOnStage(String variableValue) {
+		final String variableName = "testVariable";
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+		onView(isRoot())
+				.perform(CustomActions.wait(1500));
+
+		stageListener.makeManualScreenshot();
+		pressBack();
+		pressBack();
+
+		String filePath = Environment.getExternalStorageDirectory().getAbsolutePath();
```

Comment:
dont do this on every test.

get the correct stage contents with the untranslated value once in startup and save it in a member variable.
then you can also directly start the stage activity in the testRule. then there would be no ui interactions necessary for the whole test and no activity switches between stage and script activity